features/kafka/producer/src/test/java/org/opennms/features/kafka/producer/KafkaForwarderIT.java

```diff
@@ -420,6 +424,27 @@ public void testSameAlarmIsNotSynced() {
         }
     }
 
+
+    @Test
+    public void testNotDroppingOfMessagesWhenKafkaIsOffline() throws Exception {
+
+        eventdIpcMgr.sendNow(MockEventUtil.createNodeDownEventBuilder("test", databasePopulator.getNode1()).getEvent());
+        if (!kafkaProducer.getEventForwardedLatch().await(1, TimeUnit.MINUTES)) {
+            throw new Exception("No events were successfully forwarded in time!");
+        }
+        kafkaConsumer = startConsumer();
+        await().atMost(1, TimeUnit.MINUTES).until(this::getUeisForConsumedEvents, hasItems(
+                EventConstants.NODE_DOWN_EVENT_UEI));
+        kafkaServer.stopKafkaServer();
+        eventdIpcMgr.sendNow(MockEventUtil.createNodeUpEventBuilder("test", databasePopulator.getNode2()).getEvent());
+        // Sleep long enough to account for delivery.timeout.ms = 3 secs.
+        Thread.sleep(10000);
```

Comment:
Is there any way we can avoid sleeping for 10 seconds in this test?

Reply:
There is nothing to check for. So any other mechanism need to wait for a bit to account for `delivery.timeout.ms`